workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSComponent.cs

```diff
@@ -36,5 +36,18 @@ public bool TryGetGameObjectForSpatialOSEntityId(EntityId entityId, out GameObje
             linkedGameObject = entityManager.GetComponentObject<GameObjectReference>(entity).GameObject;
             return true;
         }
+
+        public bool TryGetSpatialOSEntityIdForGameObject(GameObject linkedGameObject, out EntityId entityId)
+        {
+            var component = linkedGameObject.GetComponent<SpatialOSComponent>();
+            if (component == null)
+            {
+                entityId = default(EntityId);
+                return false;
+            }
+
+            entityId = component.SpatialEntityId;
+            return true;
```

Comment:
If an entity is calling this from it's own SpatialOSComponent then this should also check if this is part of the same worker.

Reply:
done